__init__.py

```diff
@@ -11,1810 +11,950 @@
-        })
-        return self.to_forecast(data, "3h")
-
-    def _daily_forecast_at_location(self, name, limit):
-        if name in self.location_translations:
-            name = self.location_translations[name]
-        orig_name = name
-        while name != '':
-            try:
-                q = {"q": name}
-                if limit is not None:
-                    q["cnt"] = limit
-                data = self.request({
-                    "path": "/forecast/daily",
-                    "query": q
-                })
-                forecast = self.to_forecast(data, 'daily')
-                self.location_translations[orig_name] = name
-                return forecast
-            except HTTPError as e:
-                if e.response.status_code == 404:
-                    # Remove last word in name
-                    name = ' '.join(name.split()[:-1])
-
-        raise LocationNotFoundError('The location couldn\'t be found')
-
-    def daily_forecast(self, name, lat, lon, limit=None):
-        if lat and lon:
-            q = {"lat": lat, "lon": lon}
-        else:
-            return self._daily_forecast_at_location(name, limit)
-
-        if limit is not None:
-            q["cnt"] = limit
-        data = self.request({
-            "path": "/forecast/daily",
-            "query": q
-        })
-        return self.to_forecast(data, "daily")
-
-    def to_forecast(self, data, interval):
-        forecast = self.forecast.parse_JSON(data)
-        if forecast is not None:
-            forecast.set_interval(interval)
-            return Forecaster(forecast)
-        else:
-            return None
-
-    def set_OWM_language(self, lang):
-        self.owmlang = lang
-
-        # Certain OWM condition information is encoded using non-utf8
-        # encodings. If another language needs similar solution add them to the
-        # encodings dictionary
-        encodings = {
-            'se': 'latin1'
-        }
-        self.encoding = encodings.get(lang, 'utf8')
+from mycroft.util.parse import extract_number
+from .source import (
+    DailyWeather,
+    LocationNotFoundError,
+    OpenWeatherMapApi,
+    WeatherConfig,
+    WeatherDialog,
+    WeatherIntent,
+    WeatherReport,
+)
+
+# TODO: VK Failures
+#   Locations: Washington, D.C.
+#
+# TODO: Intent failures
+#   Later weather: only the word "later" in the vocab file works all others
+#       invoke datetime skill
+
+MARK_II = "mycroft_mark_2"
+TWELVE_HOUR = "half"
+CLEAR = 0
+PARTLY_CLOUDY = 1
+CLOUDY = 2
+LIGHT_RAIN = 3
+RAIN = 4
+THUNDERSTORM = 5
+SNOW = 6
+WINDY = 7
 
 
 class WeatherSkill(MycroftSkill):
+    """Main skill code for the weather skill."""
+
     def __init__(self):
         super().__init__("WeatherSkill")
+        self.weather_api = OpenWeatherMapApi()
+        self.weather_api.set_language_parameter(self.lang)
+        self.weather_config = WeatherConfig(self.config_core, self.settings)
+        self.platform = self.config_core["enclosure"]["platform"]
```

Comment:
Just hit an error here. The default config doesn't have the platform entry. This should probably default to "unknown" if either "enclosure" or "platform" is missing

Reply:
Fixed.

Reply:
Did you forget to push the fix, I still get the error

Reply:
should be pushed now.